index.js

```diff
@@ -0,0 +1,153 @@
+const alexa = require('ask-sdk-core')
+const skillBuilder = alexa.SkillBuilders.custom()
+const axios = require('axios')
+
+async function getDataForWeatherByCity(city) {
+    const url = `http://api.openweathermap.org/data/2.5/weather?q=${city}&APPID=${process.env.apiKey}&units=imperial`
+
+    const response = await axios.get(url)
+
+    return response.data
+
+}
+const WeatherIntentHandler = {
+    canHandle(handlerInput) {
+        return handlerInput.requestEnvelope.request.type === 'IntentRequest'
+            && handlerInput.requestEnvelope.request.intent.name === 'WeatherIntent'
+    },
+    async handle(handlerInput) {
+        const city = handlerInput.requestEnvelope.request.intent.slots.city.value
+
+        const data = await getDataForWeatherByCity(city)
+        var weatherData = data['weather']
+        var weatherDetails = ""
+        for (var i = 0; i < weatherData.length; i++) {
+            if (i > 0) {
+                weatherDetails = weatherDetails + " and "
+            }
+            weatherDetails = weatherDetails + weatherData[i].description
+        }
```

Comment:
It's a good idea to do all this in your `getDataForWeatherByCity` function so it just returns the info you need rather than having to do the parsing in your intent which should just be about responding.